pylint/pyreverse/diagrams.py

```diff
@@ -159,7 +159,9 @@ def class_names(self, nodes_lst: Iterable[nodes.NodeNG]) -> list[str]:
             if isinstance(node, astroid.Instance):
                 node = node._proxied
             if (
-                isinstance(node, (nodes.ClassDef, nodes.Name, nodes.Subscript))
+                isinstance(
+                    node, (nodes.ClassDef, nodes.Name, nodes.Subscript, nodes.BinOp)
```

Comment:
I don't think you need to add this because it adds the BinOp node to the "class name" section in the diagram?

![output](https://user-images.githubusercontent.com/31762852/170818041-2cb812e1-af01-4560-89a6-ccc8863ba778.png)

Reply:
Thank you for noticing this produces wrong output for `dot` files.

Regarding the code: it is necessary to add this, as this method is used to get the names for the names in the type annotation.
The problem was that the `|` character needs to be escaped when creating `dot` files, as it is otherwise rendered as the horizontal line we can see here between `str` and `int` for the `alternative_union_syntax` member.

It would probably be better to refactor and reuse your `get_annotation_label` here, too.
I noticed that currently the type annotations for attributes are determined in `diagrams.py`, whereas the type annotations for method signatures are determined in `printer.py`.
I added that to the list of refactoring todos (#6582).

Reply:
I think that sounds great @DudeNr33; I am sure there is room for improvements & I think your TODOs will make it a better tool for everyone. Let me know if I can help in any way & I'll be happy to pitch in when I can get time.

Reply:
Perhaps one for the TODO list also is this [issue mentioned on SO](https://stackoverflow.com/questions/71059123/pyreverse-not-showing-composition-relationships-in-the-umls-when-using-absolute).

The thing here seems to be that relationships are broken when a type-annotation uses a class imported from another module. It works ok if both classes are in the same module.

Reply:
Thank you for your offer to contribute! 😊 👍  If you'd like to take a look at unifying the way type annotations for attributes and method signatures are determined, go ahead! 

Regarding the StackOverflow question: I haven't tried it yet, but my guess is that this will be fixed in 2.14 through #6617.